### PR TITLE
Swift 3 UIKit

### PR DESCRIPTION
**Pull request by tzm41**
- Migrated some syntax to Swift 3 and corrected some formats, for #59.
- Propose to change `ip_openSettingsApp()` to throw instead of creating a `fatalError`. This might break existing code? Please give inputs.

### CODE REVIEW

**Comment by alexpersian** [comment]
Agree with the `ip_openSettingsApp()` change.

**Review by brightredchilli — APPROVED** [review]
Please wait for one more review. Thanks for the work!

File: SwiftWisdom/Core/UIKit/PercentAnimator/PercentAnimator.swift [hunk 1]
```diff
@@ -88,7 +88,7 @@ final class Animation {
         
         
         displayLink = CADisplayLink(target: self, selector: #selector(Animation.displayLinkFired))
-        displayLink.add(to: RunLoop.main, forMode: RunLoopMode.defaultRunLoopMode)
+        displayLink.add(to: RunLoop.main, forMode: .defaultRunLoopMode)
```
Comment (alexpersian) [inline SwiftWisdom/Core/UIKit/PercentAnimator/PercentAnimator.swift]: Can `RunLoop.main` be shortened to `.main` here?

**Review by alexpersian — COMMENTED** [review]
Just a small question, but the rest of this looks good.

**Pull request merged**